vendor/sigs.k8s.io/kustomize/api/internal/git/repospec.go

```diff
@@ -192,76 +246,142 @@ func peelQuery(arg string) (string, string, time.Duration, bool) {
 		}
 	}
 
-	return parsed.Path, ref, duration, submodules
+	return ref, duration, submodules
 }
 
-func parseHostSpec(n string) (string, string) {
-	var host string
-	// Start accumulating the host part.
-	for _, p := range []string{
-		// Order matters here.
-		"git::", "gh:", "ssh://", "https://", "http://",
-		"git@", "github.com:", "github.com/"} {
```

Comment:
I just spot-checked this file, but I didn't see an equivalent to this ordering... is the new code behaviorally compatible with the old code?

Reply:
(related to the above discussion https://github.com/kubernetes/kubernetes/pull/116598#discussion_r1136081750)

We did some major refactoring here but it should be behaviorally compatible as far as our documentation goes. 

From our release notes:

```
Warning: We made major improvements to the maintainability of the URL-parsing code for remote URLs, fixing several bugs in the process. We have test coverage for backwards-compatibility with all documented supported URLs, but if you experience a regression with a remote URL, please file an issue.
```

But there are two things that are not compatible:
```
https://github.com/kubernetes-sigs/kustomize/pull/4985: If you previously included .git in an AWS or Azure URL, we will no longer automatically remove that suffix. You may need to add an extra / to replace the .git for the URL to properly resolve.
https://github.com/kubernetes-sigs/kustomize/pull/4954: Drop support for using gh: as a host (e.g. gh:kubernetes-sigs/kustomize). We were unable to find any usage of or basis for this and believe it may have been targeting a custom gitconfig shorthand syntax.
```

Reply:
> https://github.com/kubernetes-sigs/kustomize/pull/4985: If you previously included .git in an AWS or Azure URL, we will no longer automatically remove that suffix. You may need to add an extra / to replace the .git for the URL to properly resolve.

will a kustomize file updated that way work properly with the previous version of kubectl / kustomize?

Reply:
Great question. We thought so when we made those changes, but I just did some tests against master and the release-4.5 branch to confirm, and it seems like there isn't in at least some cases. Seemingly this is one of the places in the old code that was not respecting the `//` delimiter and was instead relying on the invalid suffix to delineate the sub-repo path segment. 😞